src/messaging/ReliableMessageContext.h

```diff
@@ -200,16 +208,22 @@ class ReliableMessageContext
         /// When set, signifies that there is an acknowledgment pending to be sent back.
         kFlagAckPending = 0x0020,
 
+        /// When set, signifies that there has once been an acknowledgment
+        /// pending to be sent back.  In that case,
+        /// mPendingPeerAckMessageCounter is a valid message counter value for
+        /// some message we have needed to acknowledge in the past.
+        kFlagHasHadAckPending = 0x0040,
+
         /// When set, signifies that at least one message has been received from peer on this exchange context.
-        kFlagMsgRcvdFromPeer = 0x0040,
+        kFlagMsgRcvdFromPeer = 0x0080,
 
         /// When set, signifies that this exchange is waiting for a call to SendMessage.
-        kFlagWillSendMessage = 0x0080,
+        kFlagWillSendMessage = 0x00100,
```

Comment:
Nit: One too many leading zeros. Suggest `0x0100`.

Overall if you are shifting all these items, perhaps it would be better with (1u << 0), (1u << 1), ... (1u << 8) ,etc

Reply:
Done.